### DISCUSSION TITLE
Inconsistency in Trajectory Results between TF and /camera_pose Topic

### DISCUSSION THREAD

**Discussion by chuck00lin** [OpenVSLAM-Community/openvslam_ros#136] [Q&A] [answered]
Hi,

I encountered some unusual results during my tests. Specifically, when I input the transform of the robot, the SLAM trajectory appears differently. 

From the code in " stella_vslam_ros.cc", I noticed that if TF publishing is enabled, the system calculates the `map` to `odom` frame transform and then publishes this TF. In parallel, the /camera_pose topic updates it with the current camera pose.

From my understanding, the relationship between the TF and the /camera_pose topic should just be the transform matrix between the `camera_frame` and the `odom` frame.

This discrepancy has left me rather confused and I am curious if there is something I am misunderstanding, or if there is an issue within the system.

### Case 1:

```
    odom_to_base_link = Node(
        package='tf2_ros',
        executable='static_transform_publisher',
        arguments=['--x', '0', '--y', '0', '--z', '0', '--yaw', '0', '--pitch', '0', '--roll', '0',
                   '--frame-id','odom','--child-frame-id','base_link'],
        name='odom_to_base_link'
    )
```
![image](https://github.com/stella-cv/stella_vslam_ros/assets/30611555/b5f33450-17c3-47d3-b6f7-9340db3ba419)
> the trajectories are the same
---

### Case 2:
```
    odom_to_base_link = Node(
        package='tf2_ros',
        executable='static_transform_publisher',
        arguments=['--x', '0', '--y', '0', '--z', '1', '--yaw', '0', '--pitch', '0', '--roll', '0',
                   '--frame-id','odom','--child-frame-id','base_link'],
        name='odom_to_base_link'
    )
```
![image(1)](https://github.com/stella-cv/stella_vslam_ros/assets/30611555/ca6bc4ce-beef-4e3d-b5ae-1b4b9d8d5546)
> the trajectories of TF is jittering

Here are the steps that I have followed for my experiment:
```
1. Record a ROS bag that only contains the image topic.
2. Run Stella SLAM.
3. Save the log.
```

It would be great if someone could shed some light on this matter.

**Comment by ymd-stella** [comment, marked as answer]
`/camera_pose` topic is a transform from map to camera_frame (image_msg.header.frame_id). Maybe you do not have a basic understanding of tf. You need to look at tf from map to camera_frame.

**Reply by chuck00lin** [reply]
I thought the only difference between TF topic and /camera_pose should be the transform via camera_to_odom_affine. I expected that after applying a [0, 0, 1 ] transform to base_link, odom to base_link transform would only alter the Z-axis trajectory, but I'm seeing something else, X,Y trajectory are not the same.

Or could the transform formula be causing this 'jiggling' in TF? 

If this is a easy question, please let me know and i will read it myself, thanks for responding!!

![image](https://github.com/stella-cv/stella_vslam_ros/assets/30611555/c9165b6b-add3-4263-9b49-5ef55caaa875)
(this is a screenshot during salm, i attach base_link to camera, transform base_link and odom with [0,0,1] )
